src/main/java/rs/teslaris/core/service/impl/document/DocumentFileServiceImpl.java

```diff
@@ -156,6 +169,46 @@ public void parseAndIndexPdfDocument(DocumentFile documentFile, MultipartFile mu
         documentFileIndexRepository.save(documentIndex);
     }
 
+    @Override
+    public Page<DocumentFileIndex> searchDocumentFiles(SearchRequestDTO searchRequest,
+                                                       Pageable pageable, SearchRequestType type) {
+        if (type.equals(SearchRequestType.SIMPLE)) {
+            return searchService.runQuery(buildSimpleSearchQuery(searchRequest.getTokens()),
+                pageable,
+                DocumentFileIndex.class, "document_file");
+        }
+
+        return searchService.runQuery(buildAdvancedSearchQuery(searchRequest.getTokens()), pageable,
+            DocumentFileIndex.class, "document_file");
+    }
+
+    private Query buildSimpleSearchQuery(List<String> tokens) {
+        return BoolQuery.of(q -> q.must(mb -> mb.bool(b -> {
+            tokens.forEach(token -> {
+                b.should(sb -> sb.match(
+                    m -> m.field("title_sr").query(token)));
+                b.should(sb -> sb.match(
+                    m -> m.field("title_other").query(token)));
+                b.should(sb -> sb.match(
+                    m -> m.field("description_sr").query(token)));
+                b.should(sb -> sb.match(
+                    m -> m.field("description_other").query(token)));
+                b.should(sb -> sb.match(
+                    m -> m.field("pdf_text_sr").query(token)));
+                b.should(sb -> sb.match(
+                    m -> m.field("pdf_text_other")
+                        .query(token)));
+            });
+            return b;
+        })))._toQuery();
+    }
+
+    private Query buildAdvancedSearchQuery(List<String> expression) {
+        var postfixExpression =
+            expressionTransformer.transformToPostFixNotation(expression);
+        return expressionTransformer.buildQueryFromPostFixExpression(postfixExpression);
+    }
```

Comment:
This code is duplicated. Can we make this method only inside one class? Not sure about expressionTransformer, do we need separate objects for two classes for some reason?

Reply:
I managed to reduce the code to one line by encapsulating expression parsing. That way if we ever change parsing technique the responsibility will be in one single method.